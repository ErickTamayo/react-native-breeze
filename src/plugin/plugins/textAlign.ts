import { PluginFunction } from "./types";

export type PluginGroups = {
  alignment: "auto" | "left" | "right" | "center" | "justify";
};

export const pattern = /^text-(?<alignment>auto|left|right|center|justify)$/;

export const plugin: PluginFunction<PluginGroups> = ({ groups }) => {
  const { alignment } = groups;
  return { textAlign: alignment };
};
