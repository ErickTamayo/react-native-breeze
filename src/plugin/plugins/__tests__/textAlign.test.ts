import {
  generateInput,
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../helpers/tests";
import { pattern, plugin } from "../textAlign";

describe("textAlign", () => {
  // prettier-ignore
  const shouldMatch = generateInput("text", ["auto", "left", "right", "center"]);

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["text-unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
