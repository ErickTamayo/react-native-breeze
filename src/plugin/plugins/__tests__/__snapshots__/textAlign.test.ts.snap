// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textAlign Should parse (text-auto) correctly 1`] = `
Object {
  "textAlign": "auto",
}
`;

exports[`textAlign Should parse (text-center) correctly 1`] = `
Object {
  "textAlign": "center",
}
`;

exports[`textAlign Should parse (text-left) correctly 1`] = `
Object {
  "textAlign": "left",
}
`;

exports[`textAlign Should parse (text-right) correctly 1`] = `
Object {
  "textAlign": "right",
}
`;
